src/CMakeLists.txt

```diff
@@ -404,7 +404,7 @@ endif()
 CHECK_CXX_SOURCE_COMPILES("thread_local int i; int main() { i = 0; }"
 	HAVE_THREAD_LOCAL)
 
-if( NOT HAVE_THREAD_LOCAL )
+if( NOT HAVE_THREAD_LOCAL AND NOT HAIKU)
 	message( SEND_ERROR "C++ compiler doesn't support thread_local storage duration specifier" )
 endif()
```

Comment:
I don’t really get why this was changed. Thread local storage CMake test fails to build but GZDoom itself compiles and links without an issue, right? If so, the test should be fixed.

Reply:
Without this change I get this:
```CMake Error at src/CMakeLists.txt:408 (message):
  C++ compiler doesn't support thread_local storage duration specifier
```
GCC on Haiku (as of now) is compiled with thread local storage disabled due to Internal Compiler Exception (ICE).

According to https://www.doomworld.com/forum/topic/94291-haiku-os-support/?page=2&tab=comments#comment-1812795 TLS check can be ignored.

Reply:
TLS is now mandatory.

Still why does the test fail when GZDoom compiles successfully with several `thread_local` variables inside its source code?